### PR TITLE
[TT-XLA][PyTorch] DPO Training Workload for Gemma 1.1 2B Instruct - Math Preferences

### PR DESCRIPTION
**Pull request by mmilosevicTT**
### Issue
N/A

### Training Workload Setting

- **Model:** Gemma 1.1 2B Instruct
- **Strategy:** DPO + LoRA
- **Framework:** PyTorch
- **Task:** Quesiton answering
- **Dataset:** Math Preferences
- **Hardware:** QuietBox 2

### What's Changed
This PR showcases training workload for Gemma 1.1 2B Instruct. After sucessfully merging singlechip traing, this run proves we can do DPO on multichip as well.

### Experiment Results
Since DPO was run on larger sequence length, now we had to run 2 runs to confirm that the loss is not stagnating around 0.5 accuracy on validation. We can see that training continues, but validation is hardly increasing in later stages.

<table>
  <tr>
    <th>SFT Training Loss</th>
    <th>SFT Validation Loss</th>
  </tr>
  <tr>
    <td align="center">
train_loss_sft
    </td>
    <td align="center">
val_loss_sft
    </td>
  </tr>
  <tr>
    <th>DPO Training Accuracy</th>
    <th>Validation Accuracy</th>
  </tr>
  <tr>
    <td align="center">
train_acc_dpo
    </td>
    <td align="center">
val_acc_dpo
    </td>
  </tr>
  <tr>
    <th>DPO Training Reward Margin</th>
    <th>DPO Validation Reward Margin</th>
  </tr>
  <tr>
    <td align="center">
train_reward_margin_dpo
    </td>
    <td align="center">
val_reward_margin_dpo
    </td>
  </tr>
  <tr>
    <th>DPO Training Loss</th>
    <th>DPO Validation Loss</th>
  </tr>
  <tr>
    <td align="center">
train_loss_dpo
    </td>
    <td align="center">
val_loss_dpo
    </td>
  </tr>
</table>

### Checklist
- [x] Main training workflow
- [x] TT device testing

### CODE REVIEW

**Review by umalesTT — APPROVED** [review]
LGTM!
